power_grid_model_c/power_grid_model/include/power_grid_model/component/source.hpp

```diff
@@ -28,8 +28,11 @@ class Source : public Appliance {
     }
 
     explicit Source(SourceInput const& source_input, double u)
-        : Appliance{source_input, u}, u_ref_{source_input.u_ref}, u_ref_angle_{}, y1_ref_{}, y0_ref_{} {
-        u_ref_angle_ = is_nan(source_input.u_ref_angle) ? 0.0 : source_input.u_ref_angle;
+        : Appliance{source_input, u},
+          u_ref_{source_input.u_ref},
+          u_ref_angle_{is_nan(source_input.u_ref_angle) ? 0.0 : source_input.u_ref_angle},
+          y1_ref_{},
+          y0_ref_{} {
```

Comment:
since you now call the default initializer in the member definition (compile time) instead of in the constructor (runtime), you may leave leave these initializations in the constructor
```suggestion
          u_ref_angle_{is_nan(source_input.u_ref_angle) ? 0.0 : source_input.u_ref_angle} {
```

Reply:
cfr. the code smell:

> Remove this use of the constructor's initializer list for data member "y1_ref_". It is redundant with the in-class initializer.

Reply:
according to the explanation in the sonar cloud code smell:

> For C++11 or later, when you initialize a member variable in the initializer list of a constructor, but could have done so directly in the class:
> 
>     The variable has either no in-class initializer, or an in-class initializer with the same value as in the constructor
>     The initial value does not depend on a constructor parameter